autowarefoundation/autoware.universe
# fix(simple_planning_simulator): fix steering bias model

## Description

In the old implementation, `steering_bias` was just added in the measurement, thus the behavior was "the vehicle yaw dynamics totally follow the steering command, but the steering output seems different". However, the expected behavior is "the steering output seems to be ideal, but the vehicle yaw dynamics does not match its steering angle". This PR fixes the behavior.

## Related links

None

## Tests performed

1. Add `steer_bias` param in yaml. (e.g. set to 0.01)
2. Set `steering_offset.enable_auto_steering_offset_removal` to false in mpc.param.yaml
3. Run psim
4. See the vehicle trajectory following is slightly biased to the right/left.

You can also check the steering offset topic `/control/trajectory_follower/controller_node_exe/output/estimated_steer_offset` converges to the bias value if you set the offset_removal true.

## Notes for reviewers

None

## Interface changes

None

## Effects on system behavior

None by default.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.


## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.


## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.


After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/
